### PR TITLE
Time manager events

### PR DESCRIPTION
**Pull request by TomMalbran**
This is an idea proposed in #1196. Instead of having to go to WorldController and add a line to call the Update method, the TimeManager now has several events that any module can listen to, without adding extra code in the WorldManager.

Eventually we can even add Registers methods for Lua functions. But for now is just for C# code.

Since I moved some Time related code to the TimeManager, I changed a few things there.

### CODE REVIEW

**Comment by TomMalbran** [comment]
There is an error in a test. Will fix it soon.

File: Assets/Scripts/Models/World.cs [hunk 1]
```diff
@@ -159,6 +162,12 @@ public void DeleteRoom(Room r)
         r.ReturnTilesToOutsideRoom();
     }
 
+    public void AddEventListeners()
+    {
+        TimeManager.Instance.EveryFrameUnpaused += TickEveryFrame;
+        TimeManager.Instance.FixedFrequencyUnpaused += TickFixedFrequency;
```
Comment (sboigelot) [inline Assets/Scripts/Models/World.cs]: You should probably un-register this when the world get GC, we plan to have N worlds in a game session.

Comment (TomMalbran) [inline Assets/Scripts/Models/World.cs]: Good point. Will do that.

Comment (TomMalbran) [inline Assets/Scripts/Models/World.cs]: And done

**Pull request merged**